plugins/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3BlobStoreRepositoryTests.java

```diff
@@ -92,6 +111,41 @@ protected Settings nodeSettings(int nodeOrdinal) {
             .build();
     }
 
+    public void testEnforcedCooldownPeriod() throws IOException {
```

Comment:
This is admittedly quite the hacky test and it takes 2x 5s of hard waits to verify behaviour.

We could create a cleaner test by adding some BwC test infrastructure to the S3 plugin tests but I'm not sure it's worth the complexity. Also, running a real rest test to verify the timing here makes the test even more prone to run into random CI slowness and fail in the last step that verifies no waiting is happening when moving to a repo without any old version snapshot => this seemed like the least bad option to me.
We are using a O(`5s`) hard timeout in some other repo IO-timeout tests and so far that hasn't failed us due to CI running into a longer pause so I'm hopeful this will be stable.

Reply:
Not trivial but doable =>  On it :)

Reply:
Argh never mind ... then we'd have to move the cool down logic to `BlobStoreRepository`. We can't use the mock repository together with the S3 plugin. We also don't really have any mock infrastructure left for S3 so either we test this on the `BlobStoreRepository` or this requires some infrastructure that combines the mock S3 REST api infra + the snapshot resiliency test infrastructure.

Think this is worth it, given that this is a stop-gap solution?

Reply:
bummer :/